javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -530,6 +543,16 @@ public void verifyTypeScriptInstallation() {
 		extractorState.getTypeScriptParser().verifyInstallation(true);
 	}
 
+	public void extractTypeScriptFiles(List<File> files, Set<Path> extractedFiles, FileExtractor extractor) throws IOException {
+		extractorState.getTypeScriptParser().prepareFiles(files);
+		for (File f : files) {
+			Path path = f.toPath();
+			if (extractedFiles.add(path)) {
```

Comment:
If we skip a file here (because it has already been extracted), won't the TypeScript parser throw an error on the next file we request an AST for?

Reply:
Good point. It can't happen, but it's not very obvious at the moment.

The list of files given here has been built to only contain unextracted files, so the return value of `add` really doesn't need to be checked. I'll try to make this a bit more clear.

Reply:
Oh wait, there was a huge bug in the last commit here.

Reply:
Turns out the attempt to get `AutoBuildTest` working actually introduced a bug. The extracted file was added twice, and thereby skipped in the extraction step.

We'll obviously need some better testing for this file.

Reply:
I've put up an internal PR that fixes the test that should catch this.

I've confirmed that the test fails for the previous commit, and passes after 84c1ba0b31ab786785dfd3a3f40d2db15b9f9f0f.